### PR TITLE
HiKey v3

### PR DESCRIPTION
**Pull request by danh-arm**
Fixes https://github.com/ARM-software/tf-issues/issues/421.

The fastboot feature is moved to another git repository. So keep ATF only loading images & PSCI.

There are two reference files could be used to build hikey ATF/UEFI.
https://github.com/96boards-hikey/tools-images-hikey960/blob/master/build-from-source/README-ATF-UEFI-build-from-source.md
&&
https://github.com/96boards-hikey/l-loader/blob/testing/hikey960_v1.2/build_uefi.sh

Just replace the ATF git repository by upstream version.

### CODE REVIEW

**Comment by danh-arm** [comment]
One file in this PR was missing a license and the PR generally had a number of style problems. I've fixed them myself to save time.

File: plat/hisilicon/hikey/hikey_ddr.c [hunk 1]
```diff
@@ -0,0 +1,967 @@
...
+					(!(data & 0x1f0000)) || (!(data & 0x1f000000)))
+				zero_bdl = 1;
+			data = mmio_read_32((0xf712c000 + 0x214 + i * 0x80));
+			if ((!(data & 0x1f)) || (!(data & 0x1f00)) ||
+					(!(data & 0x1f0000)) || (!(data & 0x1f000000)))
+				zero_bdl = 1;
+			data = mmio_read_32((0xf712c000 + 0x218 + i * 0x80));
+			if (!(data & 0x1f))
+				zero_bdl = 1;
+			if (zero_bdl) {
```
Comment (davidcunado-arm) [inline plat/hisilicon/hikey/hikey_ddr.c]: @hzhuang1 
Static analysis check failed here - zero_bd1 can be used uninitialised. Please can you submit a fix? Thanks!

Comment (hzhuang1) [inline plat/hisilicon/hikey/hikey_ddr.c]: Fixed in https://github.com/ARM-software/arm-trusted-firmware/pull/956

**Comment by hzhuang1** [comment]
On 2017/5/30 16:50, davidcunado-arm wrote:
> *@davidcunado-arm* commented on this pull request.
> 
> ------------------------------------------------------------------------
> 
> In plat/hisilicon/hikey/hikey_ddr.c 
> <https://github.com/ARM-software/arm-trusted-firmware/pull/950#discussion_r119043248>:
> 
>> +		}
> +		mdelay(10);
> +
> +		for (i = 0; i < 4; i++) {
> +			data = mmio_read_32((0xf712c000 + 0x210 + i * 0x80));
> +			if ((!(data & 0x1f)) || (!(data & 0x1f00)) ||
> +					(!(data & 0x1f0000)) || (!(data & 0x1f000000)))
> +				zero_bdl = 1;
> +			data = mmio_read_32((0xf712c000 + 0x214 + i * 0x80));
> +			if ((!(data & 0x1f)) || (!(data & 0x1f00)) ||
> +					(!(data & 0x1f0000)) || (!(data & 0x1f000000)))
> +				zero_bdl = 1;
> +			data = mmio_read_32((0xf712c000 + 0x218 + i * 0x80));
> +			if (!(data & 0x1f))
> +				zero_bdl = 1;
> +			if (zero_bdl) {
> 
> @hzhuang1 <https://github.com/hzhuang1>
> Static analysis check failed here - zero_bd1 can be used uninitialised. 
> Please can you submit a fix? Thanks!
> 

Sure. Fixed in https://github.com/ARM-software/arm-trusted-firmware/pull/956

**Pull request merged**